### ISSUE TITLE
Pull T6 Data from google sheet

### ISSUE THREAD

**Issue by izenn** [laurinius/sto-aso#7]
**Is your feature request related to a problem? Please describe.**
Currently the t6 ships need to be manually updated/entered into data/ships.csv

**Describe the solution you'd like**
There is a community google sheet that is maintained and easy to export as CSV.  I've created a simple google sheet that looks up the values from their sheet and outputs them in the same format used by the ships.csv which then can be downloaded as a csv

Source: https://docs.google.com/spreadsheets/d/1-5Nmp_vycD2VpLbuqWnnQL1Lvx-nPy-cGYLPIckGRLk
Filtered: https://docs.google.com/spreadsheets/d/1QQL3ZictvepoSoK5Ba7eR1vwVzliKN6yS6dl4fnGWoM

**Additional context**
Next step would be to automate the retrieval of the icons.

**Comment by laurinius** [comment]
Thanks for that source. I'll probably won't add some automatic update inside the application, but instead write some script to update the data and look for discrepancies.

**Comment by laurinius** [comment]
@izenn I looked at those sheets and unfortunately the names in the lists do not match the names in the Admiralty system. E.g. in the list there the name is "Presidio Tactical Command Battlecruiser", but the correct name for the Admiralty system is "Presidio Command Battlecruiser (T6)", which makes exact matching by name difficult for an automated process.

**Comment by izenn** [comment]
yeah -- i noticed that yesterday as well.  the one thing that is consistent is the eng/tac/sci scores and the bonus -- maybe base a mapping off those?  I don't understand why they would be different, but I have a message into the google sheet maintainer about something else, hopefully they respond and i can work with them to add a field for admirality name (which would make mapping that much easier)

**Comment by laurinius** [comment]
Tried already matching based on the other values stats, unfortunately there are quite a few pairs of ships with exactly the same stats, faction, rarity etc. for everything escept the name. When looking for a partial match (like 8 out of 10 stats match), the list is even longer, and matching everything also defies the purpose of determining errors.
Also tried fuzzy logic string matching of the names, but as the overall ship naming is similar there is no reliable result either.

**Comment by izenn** [comment]
it figures...  ok, i'll let you know when Fleffle gets back to me

**Comment by laurinius** [comment]
I've written a script that allows me to add ships and icons faster by reading it from the sto.fandom.com/wiki ship page. Still needs some human input and oversight (like correct the names etc.), but much quicker than doing it all by hand.

**Comment by izenn** [comment]
if you don't mind, once you are ready, could you please publish the script/instructions in case you ever decide to move on from the project?  thanks!

**Comment by laurinius** [comment]
It is already pushed in the tools directory, it's interactive so quite explanatory, but intended to be used on the sources, not the released app (more like developer tools).
I'm also at the moment using another related script to automatically find and fill the sto-wiki url for every ship. As that URL is also part of the sheets you posted (linked on the name column) that will allow an exact match between the two, and automatic checks against the sheets and the sto wiki.

**Comment by izenn** [comment]
Awesome!  i'll leave the sheet up -- i was going to ask if you wanted it to stay around.  I'm still waiting on word back from the source maintainer because there are 3 ships with bad data (Farragut Temporal Cruiser, Courier Pilot Raider, and Terran Somerville Scout Ship) as of yesterday.  Like i said before, I'm hoping they also would be good with us adding a column for Admirality name, but i'll keep you in the loop.

**Comment by laurinius** [comment]
Still their sheet only has T6 ships, so not a full list and probably only useful as an additional source. It has a tab for small craft, but no admiralty information there.
For T1-T5 and special cards (U.S.S. Hope etc.) the only source I have, beside the game itself, is the sto wiki.

**Comment by laurinius** [comment]
Finished with the URLs for all ships, there is a separate csv mapping UUID to URLs in the resources directory. The script showed some mismatches in the values for a few entries, most of which look like typos etc. Have to check those sometime.

**Comment by laurinius** [comment]
Made my own copy of your spreadsheet @izenn , cotaining a column for the URL as text, extracted from the link:
https://docs.google.com/spreadsheets/d/1Dd5Qpyv4505Xk7edkCq8OJ4CMAlCz0rjHrE1dSY0UtE
I pulled all the data from that and added over 100 ships and icons.
Released with 1.1.3
